ompi/mca/sharedfp/lockedfile/sharedfp_lockedfile_file_open.c

```diff
@@ -99,8 +101,24 @@ int mca_sharedfp_lockedfile_file_open (struct ompi_communicator_t *comm,
         return OMPI_ERR_OUT_OF_RESOURCE;
     }
 
-    lockedfilename = (char*)malloc(sizeof(char) * (strlen(filename) + 64));
-    sprintf(lockedfilename,"%s%s",filename,".lockedfile");
+    opal_jobid_t masterjobid;
+    if ( 0 == comm->c_my_rank  ) {
+        ompi_proc_t *masterproc = ompi_group_peer_lookup(comm->c_local_group, 0 );
+        opal_process_name_t *mastername = &(masterproc->super.proc_name);
+        masterjobid = mastername->jobid;
```

Comment:
I believe the "correct" way to get the jobid involves using a macro:

```
    OMPI_CAST_RTE_NAME(&proc->super.proc_name)->jobid = jobid;
```

Other than that, this should be fine.